extensions/2.0/Khronos/KHR_materials_clearcoat/README.md

```diff
@@ -61,29 +61,58 @@ All implementations should use the same calculations for the BRDF inputs. Implem
+
+# layer coating on top of base material
+coated_material = fresnel_coat(
+  layer = clearcoat_brdf,
+  base = material,
+  ior = 1.5,
+  weight = clearcoatTexture.r * clearcoatFactor,
+  normal = clearcoatNormalTexture)
+```
+
+The `fresnel_coat` operator adds a BRDF as a layer on top of another BSDF according to `weight` and a Fresnel term. The layer is weighted with `weight*fresnel(ior)`. The base is weighted with `1-(weight*fresnel(ior)`. `normal` is a float3 vector that affects the top layer, but not the base.
+
+Implementations of the BRDF or the layering operator can vary based on device performance and resource constraints.
+
+The `clearcoatFactor` determines the view-independent intensity of the clearcoat BRDF. If `clearcoatFactor` (in the extension) is zero, the whole clearcoat layer is disabled. 
+
+Similar to the specular component of the base material, the roughness `clearcoatRoughness` is squared to make it perceptually linear. The index of refraction is set to a fixed value of 1.5, corresponding to a F0 value of 0.04.
+
+The values for clearcoat layer intensity and clearcoat roughness can be defined using factors, textures, or both. If the `clearcoatTexture` or `clearcoatRoughnessTexture` is not given, respective texture components are assumed to have a value of 1.0. All clearcoat textures contain RGB components in linear space. If both factors and textures are present, the factor value acts as a linear multiplier for the corresponding texture values.
+
+If `clearcoatNormalTexture` is not given, no normal mapping is applied to the clear coat layer, even if normal mapping is applied to the base material.  Otherwise, `clearcoatNormalTexture` may be a reference to the same normal map used by the base material, or any other compatible normal map.
+
+## Implementation
+
+*This section is non-normative.*
+
+All implementations should use the same calculations for the BRDF inputs. See [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation) for more details on the BRDF calculations. An implementation sample is available at https://github.com/KhronosGroup/glTF-Sample-Viewer/.
+
 The clearcoat formula `f_clearcoat` is computed using the specular term from the glTF 2.0 Metallic-Roughness material, defined in [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation).  Use specular F0 equal `0.04`, base color black `0.0, 0.0, 0.0`, metallic value `0.0`, and the clearcoat roughness value defined in this extension as follows:
 
 ```
 clearcoatRoughness = clearcoatRoughnessFactor * clearcoatRoughnessTexture.g
 ```
-
 The following abstract code describes how the base and clearcoat layers should be blended together:
   
 ```
 clearcoatBlendFactor = clearcoatTexture.r * clearcoatFactor
 clearcoatFresnel = fresnel(0.04, NdotV)
-
 color = (f_emissive + f_diffuse + f_specular) * (1.0 - clearcoatBlendFactor * clearcoatFresnel) +
```

Comment:
I'm wondering if this implementation section should get updated to look more like an implementation of the pseudo-code above. E.g.:
```
vec3 fresnel_coat(baseLayer, coatLayer, ior, weight)
{
    float fresnelWeight = fresnel(ior, NdotV) * weight;
    return mix(baseLayer, coatLayer, fresnelWeight);
}
```

```
vec3 baseMaterial = mix(dielectric_brdf, metal_brdf, metallic) + emission
vec3 clearcoatMaterial = specular_brdf((clearcoatRoughnessFactor * clearcoatRoughnessTexture.g)^2);  // F*Vis*D, see Appendix B
vec3 coatedMaterial = fresnel_coat(baseMaterial, clearcoatMaterial, ior, weight);
```

Reply:
Done. That's a good suggestion.